ducanhpham0312/zeevision
# feat: mock data for frontend

### Linked issue

Closes ducanhpham0312/zeevision-private#19


### Description

Backend:
- Update incorrect fields in `Process` and `Instance` type.
- Add new queries for `Instances` and `Instance`.
- Update backend README  on how to do the two points above.

Frontend:
- Implement `SingleProcessPage` page which can be navigated from `ProcessesPage`.
- Add more queried data to `ProcessesPage`.
- Remove unnecessary properties in `BpmnViewer`. Add some styling to make it looks better in `SingleProcessPage`.

### Testing

- Run `test-bench`: `./start.sh`
- Run zeevision: `docker compose up --build`
- Testing new queries: 
  - Go to the playground: `localhost:8081/playground`
  - The following query should be able to fetch all existing dummy data. Querying single process by `processKey` and single instance by `instanceKey` is also enabled.:
```
{
  processes {
    processId
    processKey
    deploymentTime
    bpmnResource
    version
    instances {
      instanceKey
      status
      startTime
    }
  }
}
```
- Frontend test: 
  - Go to `localhost:8080`, open `/processes` page. Deployment time is added
  - Clicking on a process' `Process Key` will navigate to a new page. This new page includes
    - A top-left table that include information about the process
    - The bpmn diagram of the process
    - A bottom table that includes instances' information
- Known issues (new issue tickets will be created):
  - While resizing `/processes/1` page, the BPMN diagram does not stretch accordingly. Refresh the page solves the issue but the problem occurs again if we resize the window :) Just some styling issue.
  - Compared to Figma, there should be a Slider between the first table and the diagram to toggle the width and hide the table.
  -  A Tab component to switch between process' Instances, Msg subscriptions, and Timers
  - Dummy data for msg subscriptions and timers
